lib/modules/versioning/nuget/index.spec.ts

```diff
@@ -1,77 +1,365 @@
 import nuget from '.';
 
 describe('modules/versioning/nuget/index', () => {
-  it.each`
-    input                        | expected
-    ${'9.0.3'}                   | ${true}
-    ${'1.2019.3.22'}             | ${true}
-    ${'3.0.0-beta'}              | ${true}
-    ${'2.0.2-pre20191018090318'} | ${true}
-    ${'1.0.0+c30d7625'}          | ${true}
-    ${'2.3.4-beta+1990ef74'}     | ${true}
-    ${'17.04'}                   | ${true}
-    ${'3.0.0.beta'}              | ${false}
-    ${'5.1.2-+'}                 | ${false}
-  `('isValid("$input") === $expected', ({ input, expected }) => {
-    const res = !!nuget.isValid(input);
-    expect(res).toBe(expected);
+  describe('isSingleVersion()', () => {
+    it.each`
+      input              | expected
+      ${'[1.2.3]'}       | ${true}
```

Comment:
```suggestion
      ${'[1.2.3]'}       | ${true}
      ${'1.2.3'}         | ${false}
      ${'1.2.*'}         | ${false}
```

Are these checks valid?

Reply:
What is difference between `isVersion()` and `isSingleVersion()`.

```ts
  /**
   * Check whether the `version` constraint is not a range, i.e. it only allows a
   * single specific version.
   */
  isSingleVersion(version: string): boolean;

  /**
   * Check whether the `input` is a valid version string.
   */
  isVersion(input: string | undefined | null): boolean;
```

According to these comments, there is basically no difference. However, I think `isSingleVersion()` just means a special type of range which matches exactly one version, e.g. `==1.2.3`, `=1.2.3` or `[1.2.3]`.

Reply:
Yes, you understand correctly. Another way of thinking about it is "is already pinned to a single version"